src/feature/polygon.cpp

```diff
@@ -0,0 +1,62 @@
+#include "shpxx/feature/polygon.hpp"
+
+#include "shp_casts.hpp"
+#include "shplib_helpers.hpp"
+#include "shpxx/feature/polyline_part.hpp"
+#include "shpxx/shape_type.hpp"
+
+#include <shapefil.h>
+
+namespace shpxx::feature {
+
+polygon_t::polygon_t(shplib::opaque_object_t feature) : m_underlying(std::move(feature))
```

Comment:
If these really are identical, with the difference being semantic, I'd do something like

```c++
namespace shpxx::feature {
namespace detail {
class multipart_coordinate_sequence_t {
...
};
} // namespace detail

using polygon_t = multipart_coordinate_sequence_t;
using linestring_t = multipart_coordinate_sequence_t;
}
```